tools/eslint-rules/rules/no-ngrx-fail-action-without-error-action-implementation.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * This file sets you up with structure needed for an ESLint rule.
+ *
+ * It leverages utilities from @typescript-eslint to allow TypeScript to
+ * provide autocompletions etc for the configuration.
+ *
+ * Your rule's custom logic will live within the create() method below
+ * and you can learn more about writing ESLint rules on the official guide:
+ *
+ * https://eslint.org/docs/developer-guide/working-with-rules
+ *
+ */
+
+import { ESLintUtils, TSESTree } from '@typescript-eslint/utils';
+import {
+  fixMissingImplementsInterface,
+  hasImplementsInterface,
+} from './utils/implements-interface-utils';
+import { fixPossiblyMissingImport } from './utils/import-utils';
+
+// NOTE: The rule will be available in ESLint configs as "@nrwl/nx/workspace/no-ngrx-fail-action-without-error-action-implementation"
+export const RULE_NAME =
+  'no-ngrx-fail-action-without-error-action-implementation';
+
+export const rule = ESLintUtils.RuleCreator(() => __filename)({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: ``,
+      recommended: 'recommended',
+    },
+    schema: [], // no options
+    messages: {
+      missingImplementsErrorAction:
+        '[Spartacus] NgRx Failure Action class should have `implements ErrorAction`. Otherwise it might be not handled properly by `EffectsErrorHandlerService`',
```

Comment:
[MINOR]
Outdated service name -> `ErrorActionService`
```suggestion
        '[Spartacus] NgRx Failure Action class should have `implements ErrorAction`. Otherwise it might be not handled properly by `ErrorActionService`',
```